src/base.ts

```diff
@@ -398,11 +398,16 @@ export class BaseVisitor extends AbstractVisitor<Node> {
     this.visitArguments(node.typeArguments, node.args);
   }
 
-  visitArguments(
-    typeArguments: TypeNode[] | null,
+  visitArguments(typeArguments: TypeNode[] | null, args: Expression[]): void {
+    let typeArgs = typeArguments ? typeArguments : [];
+    for(const typeArg of typeArgs) {
```

Comment:
Why not `this.visit(typeArgs)`?

Reply:
The visit sig is `visit(node: T | null): void`

Reply:
This signature?
https://github.com/willemneal/visitor-as/blob/057a447baee057fff88c432fb06050298ab08050/src/visitor.ts#L17

Reply:
You are right

Reply:
Should I also fix baseTransformer? But this leads to API breaking